### PR TITLE
add task 6

### PR DESCRIPTION
**Pull request by spiderpoul**
- [x] 1 - File serverless.yml contains configuration for catalogBatchProcess function
- [x] 2 - File serverless.yml contains policies to allow lambda catalogBatchProcess function to interact with SNS and SQS
- [x] 3 - File serverless.yml contains configuration for SQS catalogItemsQueue
- [x] 4 - File serverless.yml contains configuration for SNS Topic createProductTopic and email subscription

Additional (optional) tasks
- [ ] +1 - catalogBatchProcess lambda is covered by unit tests
- [ ] +1 - set a Filter Policy for SNS createProductTopic in serverless.yml (Create an additional email subscription and distribute messages to different emails depending on the filter for any product attribute)

### CODE REVIEW

File: product-service/serverless.yml [hunk 1]
```diff
@@ -86,54 +105,39 @@ functions:
...
-#            path: /hello
 
-#    Define function environment variables here
-#    environment:
-#      variable2: value2
+  catalogBatchProcess:
+    handler: handler.catalogBatchProcess
+    events:
+      - sqs:
+        batchSize: 2
```
Comment (KarneyenkaDzmitry) [inline product-service/serverless.yml:113]: The batch size should be 5

File: product-service/handlers/catalogBatchProcess.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            console.log('Creating product', JSON.stringify(product))
+
+            await createProduct(product);
+
+        } catch {
+            console.error(
+                "catalogBatchProcess found invalid JSON record:",
+                record.body
+            );
+            return;
```
Comment (maksym-lebedynets) [inline product-service/handlers/catalogBatchProcess.ts:48]: Return from the for...of cycle is not working as you expect.

File: product-service/handlers/catalogBatchProcess.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        }
+    }
+
+    sns.publish(
```
Comment (maksym-lebedynets) [inline product-service/handlers/catalogBatchProcess.ts:52]: It should be a part of `try` statement

File: product-service/handlers/catalogBatchProcess.ts (hunk 2 above)
Comment (spitaI) [reply product-service/handlers/catalogBatchProcess.ts:48]: \+

**Pull request merged**